### PR TITLE
docs: adding intro to sway

### PR DESCRIPTION
**Pull request by calldelegation**
Taking the readme upgrading to beta-4 and improving clarity as well as adding a rust testing section

### CODE REVIEW

**Comment by calldelegation** [comment]
TODO add intro-to-sway as submodule to pull code in. Instead of copy and pasting.

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_import"
+  commentType="//"
+  lang="rust"
+/>
+
+Always compile your contracts after making any changes. This ensures you're working with the most recent `contract-abi` that gets generated.
+
+<CodeImport
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/rust-sdk.mdx:52]: This is the entire test file, doesn't seem right.

Also I tried copying this and running it after following the steps and it doesn't work.

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  comment="rs_abi"
+  commentType="//"
+  lang="rust"
+/>
+
+## Initializing Functions
+
+When writing tests for Sway, two crucial objects are required: the contract instance and the wallets that interact with it. This helper function ensures a fresh start for every new test case. It will export the deployed contracts, the contract ID, and all the generated wallets for this purpose.
+
+<CodeImport
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/rust-sdk.mdx:63]: It's unclear if we are supposed to copy and paste this or not. It's different that what is generated. I would take a look at the Fuel Journey Rust guide to see if that helps with outlining this.

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+  lang="rust"
+/>
+
+## Test Cases
+
+Given the immutable nature of smart contracts, it's not only vital to test basic functionalities but also imperative to thoroughly cover all potential edge cases.
+
+### Setting Owner
+
+For this test case, we utilize the contract instance and harness the SDK's `.with_account()` method. This method allows us to mimic the actions of the first wallet, especially the 'initialize owner' procedure. To ascertain the accuracy of the owner designation, we can confirm that the address returned by the contract aligns with that of wallet 1. For a more in-depth verification, probing the contract storage will reveal whether wallet 1's address is appropriately stored.
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/rust-sdk.mdx:87]: "This method allows us to mimic the actions of the first wallet, especially the 'initialize owner' procedure." - This sentence is confusing.

Also, the tone of rest of the paragraph feels slightly too formal/academic.

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_contract_instance_config"
+  commentType="//"
+  lang="rust"
+/>
+
+## Test Cases
+
+Given the immutable nature of smart contracts, it's not only vital to test basic functionalities but also imperative to thoroughly cover all potential edge cases.
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/rust-sdk.mdx:83]: Clarify that in this section the user should be adding these functions to the bottom of harness.rs

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+```bash
+cargo install cargo-generate
+```
+
+2. Generate the template:
+```bash
+cargo generate --init fuellabs/sway templates/sway-test-rs --name contract
+```
+
+### Running the Tests
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/rust-sdk.mdx:27]: This section should be at the end of this page

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+To create your own test template using Rust, follow these steps with `cargo-generate`:
+
+1. Install `cargo-generate`:
+```bash
+cargo install cargo-generate
+```
+
+2. Generate the template:
+```bash
+cargo generate --init fuellabs/sway templates/sway-test-rs --name contract
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/rust-sdk.mdx:24]: This generates a test with fuels-rs 0.47.0, but your test uses 0.48.0

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+
+```shell
+npx create-react-app frontend --template typescript
+```
+
+Next, install the fuels Typescript and wallet SDKs in the frontend folder and generate types from your contract with `fuels typegen`.
+
+```shell
+cd frontend
+npm install fuels @fuel-wallet/sdk
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:23]: we need to specify versions for both of these like in the quickstart

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+npx fuels typegen -i ../contract/out/debug/*-abi.json -o ./src/contracts
+```
+
+In the `tsconfig.json` file, add the line below in the `compilerOptions` object to add the Fuel wallet type on the window object.
+
+```json
+"types": ["@fuel-wallet/sdk"],
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:30]: I don't think this is needed anymore

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+  file="../../../intro-to-sway/frontend/src/components/AllItems.tsx"
+  comment="fe_all_items_state_variables"
+  commentType="//"
+  lang="tsx"
+/>
+
+Next, let's fetch the items in a `useEffect` hook.
+Because these are read-only functions, we can simulate a dry-run of the transaction by using the `get` method instead of `call` so the user doesn't have to sign anything.
+
+<CodeImport
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:244]: This is a change for the intro-to-sway repo, but there is a new method in the ts sdk to convert to a number, we should use this instead of my work-around `parseFloat(value.format()) * 1_000_000_000;`

File: docs/guides/docs/intro-to-sway/explore-fuel.mdx [hunk 10]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/explore-fuel.mdx:1]: I think we can remove this page, everything is linked in the nav already

**Review by sarahschwartz — CHANGES_REQUESTED** [review]

File: docs/guides/docs/intro-to-sway/explore-fuel.mdx (hunk 10 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/explore-fuel.mdx:1]: agreed. nuking it

File: docs/guides/docs/intro-to-sway/prerequisites.mdx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+title: Prerequisites
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
+## Prerequisites
+
+Before diving into any Sway code, ensure you have installed the following dependencies.
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/prerequisites.mdx:11]: I added the link https://docs.fuel.network/guides/installation/. Waiting for this to merge https://github.com/FuelLabs/docs-hub/pull/62

File: docs/guides/docs/intro-to-sway/prerequisites.mdx [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+You can check to see the current toolchain version installed by running the following:
+
+```bash
+$ fuelup show
+```
+
+Next, add the [Sway extension](https://marketplace.visualstudio.com/items?itemName=FuelLabs.sway-vscode-plugin) to your VS Code.
+
+If you want to be able to run tests in Rust, install the [Rust toolchain](https://www.rust-lang.org/tools/install) as well.
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/prerequisites.mdx:30]: Yes good point removed the entire section for @cold-briu 's new installation guide

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx (hunk 9 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/typescript-sdk.mdx:244]: fixed here https://github.com/FuelLabs/intro-to-sway/pull/21
- [ ] update docs
- [ ] pull in new intro-to-sway here

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx (hunk 5 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/rust-sdk.mdx:27]: Good call

File: docs/guides/docs/intro-to-sway/contract-imports.mdx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+<CodeImport
+  file="../../../intro-to-sway/contract/src/docs_hub_misc.sw"
+  comment="import_multi"
+  commentType="//"
+  lang="sway"
+/>
+
+For this contract, here is what needs to be imported:
+
+<CodeImport
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/contract-imports.mdx:41]: changes reflected in this [PR](https://github.com/FuelLabs/intro-to-sway/pull/21)

File: docs/guides/docs/intro-to-sway/contract-structs.mdx [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+`u64`: a 64-bit unsigned integer.
+
+In Sway, there are four native types of numbers:
+- `u8`: an 8-bit unsigned integer.
+- `u16`: a 16-bit unsigned integer.
+- `u32`: a 32-bit unsigned integer.
+- `u64`: a 64-bit unsigned integer.
+
+An unsigned integer means there is no `+` or `-` sign, making the value always positive. `u64` is the default type used for numbers in Sway. To use other number types, such as [`u256`](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/u256.sw) or [signed integers](https://github.com/FuelLabs/sway-libs/tree/master/sway_libs/src/signed_integers), you must import them from a library.
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/contract-structs.mdx:38]: didn't know this thanks!

File: docs/guides/docs/intro-to-sway/contract-functions.mdx [hunk 15]
```diff
@@ -0,0 +1,315 @@
...
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="buy_item_require_update_storage"
+  commentType="//"
+  lang="sway"
+/>
+
+### Transferring payment
+
+Lastly, we can process the payment to the seller. It's recommended to transfer assets only after all storage modifications are completed to prevent [re-entrancy attacks](https://fuellabs.github.io/sway/v0.32.1/book/blockchain-development/calling_contracts.html#handling-re-entrancy).
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/contract-functions.mdx:178]: https://docs.fuel.network/docs/sway/blockchain-development/calling_contracts/#handling-re-entrancy
Using the new docs hub link

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx [hunk 16]
```diff
@@ -0,0 +1,323 @@
...
+title: Rust Testing
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
+## Testing the contract
+
+You can find the full contract code, along with example tests using the Rust SDK, in the provided repository.
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/rust-sdk.mdx:11]: You're right. I just removed it.

File: docs/guides/docs/intro-to-sway/checkpoint.mdx [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+
+Post-compilation, the system will automatically generate `abi.json`, `storage_slots.json`, and `contract.bin`. You can locate these files in the following directory:
+```
+contract/out/debug/*
+```
+
+## Deploying the contract
+
+For detailed steps on deploying this contract, refer to the official Fuel developer quickstart guide: 
+[Deploy the Contract](https://fuelbook.fuel.network/master/quickstart/smart-contract.html#deploy-the-contract)
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/checkpoint.mdx:43]: done

File: docs/guides/docs/intro-to-sway/contract-functions.mdx [hunk 18]
```diff
@@ -0,0 +1,315 @@
...
+
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="get_item"
+  commentType="//"
+  lang="sway"
+/>
+
+
+To return a value in a function, you can use the `return` keyword, similar to JavaScript. Alternatively, you can omit the semicolon in the last line to return that value. Although both methods are effective, it's generally better to be explicit.
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/contract-functions.mdx:207]: I will remove "it's generally better to be explicit." but I'm going to leave returns in the examples because this detail can be easily overlooked if you are coming from Javascript and not familar with rust.

File: docs/guides/docs/intro-to-sway/contract-functions.mdx [hunk 19]
```diff
@@ -0,0 +1,315 @@
...
+
+Should the amount exceed 0.1 ETH, a commission is determined and then deducted from the total.
+
+To facilitate the payment to the item's owner, the `transfer` function is utilized. This function, sourced from the standard library, requires three parameters: the coin quantity for transfer, the coin's asset ID, and the Identity to which the coins are dispatched.
+
+## 3. Get an item
+
+To get the details for an item, we can create a read-only function that returns the `Item` struct for a given item ID.
+
+<CodeImport
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/contract-functions.mdx:201]: It does in Rust but not in Sway. The code builds fine. I will still add it though just to be consistent.

File: docs/guides/docs/intro-to-sway/contract-structs.mdx [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+
+In JavaScript, there are two types of integers: `number` and `BigInt`. The primary difference between these types is that `BigInt` can store much larger values. Similarly, each numeric type in Sway has its maximum value that can be stored.
+
+`String`: a string created from the string type in the standard library. If you want to use the `str` keyword, you must define a fixed length, and it cannot be changed.
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/contract-structs.mdx:43]: Yes good call this is new

File: docs/guides/docs/intro-to-sway/contract-abi.mdx [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="abi"
+  commentType="//"
+  lang="sway"
+/>
+
+### Functions
+
+A function is defined using the `fn` keyword. In Sway, snake case is the convention, so instead of naming a function `myFunction`, you would name it `my_function`.
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/contract-abi.mdx:24]: We already go over that in the functions part. I will add a note for the reader to not worry about what each of the functions do as we will be explaining that when we write out the contents.

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx (hunk 2 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/rust-sdk.mdx:49]: This includes the changes from rust sdk v0.48.0 so its an updated version of what is generated. I added a note in there to help clear up that the user needs to copy this in.

File: docs/guides/docs/intro-to-sway/rust-sdk.mdx (hunk 1 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/rust-sdk.mdx:38]: Should work now

File: docs/guides/docs/guides.json [hunk 22]
```diff
@@ -1,18 +1,22 @@
...
-  }
-}
+{
+  "installation": {
+    "title": "Toolchain Installation",
+    "description": "Install the Fuel toolchain and binaries."
+  },
+  "quickstart": {
+    "title": "Developer Quickstart",
+    "description": "Get started by building a counter dApp in Fuel"
```
Comment (cold-briu) [inline docs/guides/docs/guides.json:8]:
```suggestion
    "description": "Get started by building a counter dApp in Fuel."
```
Adding dot at the end of the description.

File: docs/guides/docs/intro-to-sway/checkpoint.mdx [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+title: Checkpoint
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
+## Sway Contract Checkpoint
+
+If you have followed the previous steps correctly your `main.sw` marketplace contract should look like this
```
Comment (cold-briu) [inline docs/guides/docs/intro-to-sway/checkpoint.mdx:11]:
```suggestion
If you have followed the previous steps correctly your `main.sw` marketplace contract should look like this:
```
Adding a colon at the end.

File: docs/guides/docs/intro-to-sway/checkpoint.mdx [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+  commentType="//"
+  lang="sway"
+/>
+
+## Building the contract
+
+Here's a polished version of your instructions:
+
+To format your contract, execute the command:
+```sh
```
Comment (cold-briu) [inline docs/guides/docs/intro-to-sway/checkpoint.mdx:25]: Linter might require to add line breaks before and after the code blocks.

File: docs/guides/docs/intro-to-sway/checkpoint.mdx [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+
+Post-compilation, the system will automatically generate `abi.json`, `storage_slots.json`, and `contract.bin`. You can locate these files in the following directory:
+```
+contract/out/debug/*
+```
+
+## Deploying the contract
+
+For detailed steps on deploying this contract, refer to the official Fuel developer quickstart guide: 
+[Deploy the Contract](https://docs.fuel.network/guides/quickstart/building-a-smart-contract/#deploy-the-contract)
```
Comment (cold-briu) [inline docs/guides/docs/intro-to-sway/checkpoint.mdx:43]: Should we use relative links instead of DNS? 🤔

File: docs/guides/docs/intro-to-sway/checkpoint.mdx [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+
+To deploy, use the following command if you've already set up the forc-wallet and have testnet funds in your account. If not, please follow the instructions above.
+
+```sh
+forc-deploy --node-url https://beta-4.fuel.network/graphql --gas-price 1 --gas-limit 1000000
```
Comment (cold-briu) [inline docs/guides/docs/intro-to-sway/checkpoint.mdx:48]:
```suggestion
forc deploy -testnet```

I think this is more elegant and does the same job.

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 27]
```diff
@@ -0,0 +1,344 @@
...
+---
+
+## Building the Frontend
+
+### Setup
+
+Initialize a new React app with TypeScript in the same parent folder as your contract using the command below.
+
+```shell
+npx create-react-app frontend --template typescript
```
Comment (cold-briu) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:16]: I understand that the example was already generated using CRA. However, should we move forward into using vite?

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 28]
```diff
@@ -0,0 +1,344 @@
...
+At the top of the file, add your contract ID as a constant.
+
+<CodeImport
+  file="../../examples/intro-to-sway/frontend/src/App.tsx"
+  comment="fe_contract_id"
+  commentType="//"
+  lang="tsx"
+/>
+
+Next, create a new folder in the `src` folder called `hooks`, and copy and paste the `useFuel.tsx` and `useIsConnected.tsx` hooks from the example in this repo. You can also find them in the [offical wallet docs](https://wallet.fuel.network/docs/react/).
```
Comment (cold-briu) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:56]: I would change this section, since the hooks can be found at @fuel-wallet/react.

More details [here](https://github.com/FuelLabs/fuels-wallet/blob/docs/fuel-wallet-react/packages/docs/docs/dev/getting-started.mdx#L84C17-L84C17).

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 29]
```diff
@@ -0,0 +1,344 @@
...
+
+```shell
+npx create-react-app frontend --template typescript
+```
+
+Next, install the fuels Typescript and wallet SDKs in the frontend folder and generate types from your contract with `fuels typegen`.
+
+```shell
+cd frontend
+npm install fuels@0.60.0 @fuel-wallet/sdk@0.13.0 --save
```
Comment (cold-briu) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:23]: Include  `@fuel-wallet/react`

File: docs/guides/docs/intro-to-sway/checkpoint.mdx (hunk 25 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/checkpoint.mdx:43]: Good call on this

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx (hunk 27 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/typescript-sdk.mdx:16]: I think this is fine for now

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx (hunk 28 above)
Comment (calldelegation) [reply docs/guides/docs/intro-to-sway/typescript-sdk.mdx:56]: Yup this is outdated good catch

**Comment by calldelegation** [comment]
Updating testing template here https://github.com/FuelLabs/sway/pull/5181

File: src/lib/plugins/code-import.ts [hunk 30]
```diff
@@ -8,10 +8,15 @@ import path from 'path';
 import * as prettier from 'prettier';
 import type { Root } from 'remark-gfm';
 import { visit } from 'unist-util-visit';
-
 import { getEndCommentType } from './text-import';
```
Comment (cold-briu) [inline src/lib/plugins/code-import.ts:11]:
```suggestion
```
should remove this

**Review by cold-briu — CHANGES_REQUESTED** [review]
code-import.ts requires a couple fixes to pass the linter test

File: docs/guides/docs/intro-to-sway/prerequisites.mdx [hunk 31]
```diff
@@ -0,0 +1,154 @@
...
+  lang="sh"
+  trim="true"
+/>
+
+<TextImport
+  file="../quickstart/building-a-smart-contract.mdx"
+  comment="install_help"
+  commentType="{/*"
+/>
+
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/prerequisites.mdx:72]: Another prerequisite is node/npm 18.14.1 (or above up to node 19).

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 32]
```diff
@@ -0,0 +1,697 @@
...
+/>
+
+<CodeImport
+  file="../../examples/intro-to-sway/frontend/src/index.tsx"
+  comment="fe_index_all"
+  commentType="//"
+  lang="tsx"
+/>
+
+At the top of the file, add your contract ID that you had earlier as a constant.
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:114]: This is out of place after editing the `index.tsx` file. I think it makes sense to edit `index.tsx` and then `App.tsx`.

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 33]
```diff
@@ -0,0 +1,697 @@
...
+
+<CodeImport
+  file="../../examples/intro-to-sway/frontend/src/App.tsx"
+  comment="fe_contract_id"
+  commentType="//"
+  lang="tsx"
+/>
+
+Webhooks from the Fuel Wallet package are used in order to connect our wallet to the dapp. In the `App` function, we can call these hooks like this:
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:123]:
```suggestion
React hooks from the Fuel Wallet package are used in order to connect our wallet to the dapp. In the `App` function, we can call these hooks like this:
```

File: docs/guides/docs/intro-to-sway/typescript-sdk.mdx [hunk 34]
```diff
@@ -0,0 +1,697 @@
...
+
+<CodeImport
+  file="../../examples/intro-to-sway/frontend/src/App.tsx"
+  comment="fe_call_hooks"
+  commentType="//"
+  lang="tsx"
+/>
+
+The state variable called `wallets` with the `useState` hook will have the type `WalletLocked`.
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/typescript-sdk.mdx:132]:
```suggestion
The `wallet` variable from the `useWallet` hook will have the type `FuelWalletLocked`.
```

File: docs/guides/docs/intro-to-sway/explore-fuel.mdx [hunk 35]
```diff
@@ -0,0 +1,29 @@
...
+title: Explore Fuel
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
+## Keep building on Fuel
+
+Ready to keep building? You can dive deeper into Sway and Fuel in the resources below:
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/explore-fuel.mdx:11]: We should ask people to tweet about the dapp like we do in the quickstart

**Review by sarahschwartz — CHANGES_REQUESTED** [review]
Looking really good! I have some mostly small changes, and then I think it's ready 🌴

File: tests/utils/checks.ts [hunk 36]
```diff
@@ -15,6 +15,7 @@ export function checkIfIsIncremented(initialIndex: number, finalIndex: number) {
   console.log('INITIAL:', initial);
   console.log('FINAL:', final);
   const isIncremented = final === initial + 1;
+  saved.length = 0;
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply tests/utils/checks.ts:18]: This resets the saved variable. Before the state inside saved would be carried through onto each test rendering it incorrect. 

i.e. 
before: [123, 124, 2, 3]

where 123, 124 is the quickstart test and 2, 3 is the intro to sway test. Since the function always looks at the first two in the array, tests will always run true if they run after quickstart. 

after: [2, 3]

File: tests/test.spec.ts [hunk 37]
```diff
@@ -24,4 +24,39 @@ test.describe('Guides', () => {
     stopServers();
     context.close();
   });
+
```
[the remark this answers is not in the document]
Comment (calldelegation) [reply tests/test.spec.ts:27]: Added parallelization as well so the CI runs faster 😄

File: tests/utils/checks.ts (hunk 36 above)
[the remark this answers is not in the document]
Comment (sarahschwartz) [reply tests/utils/checks.ts:18]: Can you change this to `saved = [];` and use `let` to declare the save array instead?

File: docs/guides/docs/guides.json [hunk 38]
```diff
@@ -11,6 +11,12 @@
     "featured": true,
     "tags": ["Getting Started", "Full Stack"]
   },
+  "intro_to_sway": {
+    "title": "Introduction to Sway",
+    "description": "Learn Sway fundamentals by building a marketplace dApp.",
+    "featured": false,
+    "tags": []
```
Comment (sarahschwartz) [inline docs/guides/docs/guides.json:18]: Can you add the full-stack tag here?

File: docs/guides/docs/guides.json [hunk 39]
```diff
@@ -11,6 +11,12 @@
     "featured": true,
     "tags": ["Getting Started", "Full Stack"]
   },
+  "intro_to_sway": {
+    "title": "Introduction to Sway",
```
Comment (sarahschwartz) [inline docs/guides/docs/guides.json:15]: I think we should keep the "for javascript developers" in the title here

File: docs/guides/docs/intro-to-sway/index.mdx [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+# Introduction to Sway Language for JavaScript Developers
+
+If you're familiar with JavaScript and have a basic understanding of blockchain fundamentals, you can swiftly grasp how to build full-stack decentralized applications on Fuel using Sway. Once you get a handle on Sway's essentials, you'll be able to begin building your own dapp.
+
+Within this tutorial, we will be crafting a Sway contract for an online marketplace similar to Amazon, where:
+
+1. Sellers can list products.
+2. Buyers can purchase those products.
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/index.mdx:16]: We should add a preview image for what the app looks like.

File: docs/guides/docs/intro-to-sway/checkpoint.mdx [hunk 41]
```diff
@@ -0,0 +1,71 @@
+---
+title: Checkpoint
+category: Intro to Sway
+parent:
+  label: Guides
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/checkpoint.mdx:5]: Can you change all of the parent labels to "All Guides" to match the other guides?

File: docs/guides/docs/intro-to-sway/prerequisites.mdx [hunk 42]
```diff
@@ -0,0 +1,165 @@
+---
+title: Prerequisites
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
```
Comment (sarahschwartz) [inline docs/guides/docs/intro-to-sway/prerequisites.mdx:8]: Can you make sure each page has an h1 header (like this page could have `# Prerequisites`)

**Review by sarahschwartz — CHANGES_REQUESTED** [review]
almost there, just a few more nits

**Review by sarahschwartz — APPROVED** [review]
LGTM, great job!! 👏

File: docs/about-fuel/networks/beta-3.md
Comment (cold-briu) [inline docs/about-fuel/networks/beta-3.md:1]: Consider implementing the dynamic {{testnet.version}} approach designed by @sarahschwartz?

File: docs/guides/docs/installation/index.mdx [hunk 43]
```diff
@@ -51,13 +51,13 @@ This will install `forc`, `forc-client`, `forc-fmt`, `forc-lsp`, `forc-wallet` a
 
 👉 Just paste the following line in your terminal and press *Enter*.
 
-{/*install_fuelup_command:example:start*/}
+{/*ANCHOR: install_fuelup_command*/}
 
 ```sh
 curl --proto '=https' --tlsv1.2 -sSf https://install.fuel.network/fuelup-init.sh | sh
```
Comment (cold-briu) [inline docs/guides/docs/installation/index.mdx:57]: Please consider that "curl -sSf https://install.fuel.network/fuelup-init.sh | sh" work as expected since: 

-  The url scheme definition should be enough to enforce the HTTPS protocol 
- Curl will negotiate the highest SSL supported by the server.
- It seems that --proto '=https' --tlsv1.2 can be removed right away.

I would love to hear your thoughts on this.

**Review by cold-briu — APPROVED** [review]
Great job! 
Thanks for remove the `$` characters. 

I've dropped a couple of non-blocking comments.

File: docs/about-fuel/networks/beta-3.md
Comment (calldelegation) [reply docs/about-fuel/networks/beta-3.md:1]: This goes along with the comment below, will be addressed in a separate PR

File: docs/guides/docs/installation/index.mdx (hunk 43 above)
Comment (calldelegation) [reply docs/guides/docs/installation/index.mdx:57]: Good point for sure but I don't think this is relevant in this PR. Will address this separately :)
